src/engines/kotor/gui/guibackground.cpp

```diff
@@ -0,0 +1,142 @@
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  The menu background.
+ */
+
+#include <cfloat>
+
+#include "src/common/util.h"
+#include "src/common/ustring.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/types.h"
+
+#include "src/graphics/graphics.h"
+#include "src/graphics/resolution.h"
+#include "src/graphics/aurora/textureman.h"
+
+#include "src/engines/kotor/gui/guibackground.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+GuiBackground::GuiBackground(const Common::UString &type) :
+	Graphics::GUIElement(Graphics::GUIElement::kGUIElementBack), _type(type) {
+
+	_distance = FLT_MAX;
+
+	_screenWidth = GfxMan.getScreenWidth();
+	_screenHeight = GfxMan.getScreenHeight();
+
+	update();
+}
+
+GuiBackground::~GuiBackground() {
+	hide();
+}
+
+void GuiBackground::setType(const Common::UString &type) {
+	_type = type;
+}
+
+void GuiBackground::calculateDistance() {
+}
+
+void GuiBackground::render(Graphics::RenderPass pass) {
+	if (pass == Graphics::kRenderPassTransparent || !_render)
+		return;
+
+	TextureMan.set(_texture);
+
+	glDisable(GL_DEPTH_TEST);
+	glDepthMask(GL_FALSE);
+
+	glMatrixMode(GL_PROJECTION);
+	glLoadIdentity();
+	glOrtho(0.0, _screenWidth, 0.0, _screenHeight, -1.0, 1.0);
+
+	glMatrixMode(GL_MODELVIEW);
+	glLoadIdentity();
+
+	glPushMatrix();
+
+	glBegin(GL_QUADS);
+		glTexCoord2f(0.0, 0.0);
+		glVertex2i(_vertexX1, _vertexY1);
+		glTexCoord2f(1.0, 0.0);
+		glVertex2i(_vertexX2, _vertexY1);
+		glTexCoord2f(1.0, 1.0);
+		glVertex2i(_vertexX2, _vertexY2);
+		glTexCoord2f(0.0, 1.0);
+		glVertex2i(_vertexX1, _vertexY2);
+	glEnd();
+
+	glPopMatrix();
+
+	glDepthMask(GL_TRUE);
+	glEnable(GL_DEPTH_TEST);
+}
+
+void GuiBackground::update() {
+	if (tryBackground(_screenWidth, _screenHeight)) {
+		return;
+	}
+
+	for (size_t i = 0; i < Graphics::resolutions.size(); i++) {
+		if (Graphics::resolutions[i].width <= _screenWidth && Graphics::resolutions[i].height <= _screenHeight) {
+			if (tryBackground(Graphics::resolutions[i].width, Graphics::resolutions[i].height)) {
+				return;
+			}
+		}
+	}
+}
+
+bool GuiBackground::tryBackground(int width, int height) {
+	Common::UString name = Common::UString::format("%ux%u%s", width, height, _type.c_str());
+	if (ResMan.hasResource(name, Aurora::kResourceImage)) {
+		_texture = TextureMan.get(name);
+
+		_vertexX1 = (_screenWidth - width) / 2;
+		_vertexX2 = _vertexX1 + width;
+		_vertexY1 = (_screenHeight - height) / 2;
+		_vertexY2 = _vertexY1 + height;
+
+		_render = true;
+		status("found background %s", _texture.getName().c_str());
```

Comment:
Minor nitpick, but isn't it a bit too spammy for status()? Maybe put it in a debugC() instead?